buildutils/detect.py

```diff
@@ -58,26 +58,21 @@ def test_compilation(cfile, compiler=None, **compiler_attrs):
     cc.link_executable(objs, efile, extra_preargs=lpreargs)
     return efile
 
-def compile_and_run(basedir, src, compiler=None, **compiler_attrs):
+def compile_and_forget(basedir, src, compiler=None, **compiler_attrs):
+    """Make sure src compiles and links successfully.
+
+    The resulting binary is deleted without being run.
+    """
     if not os.path.exists(basedir):
         os.makedirs(basedir)
     cfile = pjoin(basedir, os.path.basename(src))
     shutil.copy(src, cfile)
     try:
         cc = get_compiler(compiler, **compiler_attrs)
         efile = test_compilation(cfile, compiler=cc)
-        patch_lib_paths(efile, cc.library_dirs)
```

Comment:
They're never needed though: you don't need to run the code to test for "sys/un.h", it's purely a compile thing.

IIRC previously `compile_and_run` was used to detect the ZMQ library version, but now `detect_zmq` calls `test_compilation` directly. I think this code is simply a leftover from that time, it was never needed for the sys/un test.